great_expectations/datasource/data_connector/runtime_data_connector.py

```diff
@@ -51,9 +54,80 @@ def __init__(
             batch_spec_passthrough=batch_spec_passthrough,
         )
 
-        self._batch_identifiers = batch_identifiers
+        if assets is None:
+            assets = {}
+        self._assets: Dict[str, Union[dict, Asset]] = assets
+        self._batch_identifiers: dict = {}
+
+        self._build_assets_from_config(config=assets)
+
+        # add batch_identifiers defined at the DataConnector level.
+        self._add_batch_identifiers(batch_identifiers)
         self._refresh_data_references_cache()
 
+    @property
+    def assets(self):
+        return self._assets
+
+    def _build_assets_from_config(self, config: Dict[str, dict]):
+        """
+        Read in asset configurations from RuntimeDataConnector. Build and load into assets property, and load
+        batch_identifiers.
+
+        Args:
+            config (dict): Asset configurations at the DataConnector-level
+        """
+        for name, asset_config in config.items():
```

Comment:
@Shinnnyshinshin Any chance to declare type hints above for `name` and `asset_config`?  Thanks!

Reply:
done 🙇🏼